patches/minecraft/net/minecraft/world/gen/settings/DimensionGeneratorSettings.java.patch

```diff
@@ -1,6 +1,73 @@
+          return p_236213_0_.field_236209_i_;
+-      })).apply(p_236214_0_, p_236214_0_.stable(DimensionGeneratorSettings::new));
++      }), net.minecraftforge.common.ForgeHooks.ALL_DIMENSIONS_CODEC.forGetter(dgs -> dgs.forgeAllDimensionsTag)).apply(p_236214_0_, p_236214_0_.stable(DimensionGeneratorSettings::new));
+    }).comapFlatMap(DimensionGeneratorSettings::func_236233_n_, Function.identity());
+    private static final Logger field_236203_c_ = LogManager.getLogger();
+    private final long field_236205_e_;
+@@ -52,6 +52,7 @@
+    private final boolean field_236207_g_;
+    private final SimpleRegistry<Dimension> field_236208_h_;
+    private final Optional<String> field_236209_i_;
++   private final net.minecraft.nbt.CompoundNBT forgeAllDimensionsTag;
+ 
+    private DataResult<DimensionGeneratorSettings> func_236233_n_() {
+       Dimension dimension = this.field_236208_h_.func_230516_a_(Dimension.field_236053_b_);
+@@ -68,6 +69,9 @@
+ 
+    public DimensionGeneratorSettings(long p_i231914_1_, boolean p_i231914_3_, boolean p_i231914_4_, SimpleRegistry<Dimension> p_i231914_5_) {
+       this(p_i231914_1_, p_i231914_3_, p_i231914_4_, p_i231914_5_, Optional.empty());
++   }
++   public DimensionGeneratorSettings(long p_i231914_1_, boolean p_i231914_3_, boolean p_i231914_4_, SimpleRegistry<Dimension> p_i231914_5_, net.minecraft.nbt.CompoundNBT forgeAllDimensionsTag) {
++      this(p_i231914_1_, p_i231914_3_, p_i231914_4_, p_i231914_5_, Optional.empty(), forgeAllDimensionsTag);
+       Dimension dimension = p_i231914_5_.func_230516_a_(Dimension.field_236053_b_);
+       if (dimension == null) {
+          throw new IllegalStateException("Overworld settings missing");
+@@ -75,11 +79,15 @@
+    }
+ 
+    private DimensionGeneratorSettings(long p_i231915_1_, boolean p_i231915_3_, boolean p_i231915_4_, SimpleRegistry<Dimension> p_i231915_5_, Optional<String> p_i231915_6_) {
++      this(p_i231915_1_, p_i231915_3_, p_i231915_4_, p_i231915_5_, p_i231915_6_, new net.minecraft.nbt.CompoundNBT());
++   }
++   private DimensionGeneratorSettings(long p_i231915_1_, boolean p_i231915_3_, boolean p_i231915_4_, SimpleRegistry<Dimension> p_i231915_5_, Optional<String> p_i231915_6_, net.minecraft.nbt.CompoundNBT forgeAllDimensionsTag) {
+       this.field_236205_e_ = p_i231915_1_;
+       this.field_236206_f_ = p_i231915_3_;
+       this.field_236207_g_ = p_i231915_4_;
+       this.field_236208_h_ = p_i231915_5_;
+       this.field_236209_i_ = p_i231915_6_;
++      this.forgeAllDimensionsTag = forgeAllDimensionsTag;
```

Comment:
Setting this at the top of the constructor instead of the bottom would make this a smaller patch/reduce the hunks.